js_modules/dagster-ui/packages/ui-core/src/sensors/SensorSwitch.tsx

```diff
@@ -61,6 +91,74 @@ export const SensorSwitch = (props: Props) => {
 
   const running = status === InstigationStatus.RUNNING;
 
+  if (
+    !running &&
+    canStartSensor &&
+    sensor.sensorType === SensorType.RUN_STATUS &&
+    _isRunStatusSensorLagging(cursor)
+  ) {
+    // We are turning back on a sensor that has a persisted cursor for a position far back in time.
+    // We should warn the user that this is what is happening.
+    const last_processed_timestamp = _parseRunStatusSensorCursor(cursor);
+    return (
+      <>
+        <Checkbox
+          format="switch"
+          disabled={toggleOnInFlight || toggleOffInFlight}
+          checked={running || toggleOnInFlight}
+          onChange={() => setShowRunStatusWarningDialog(true)}
+          size={size}
+        />
+        <Dialog
+          isOpen={showRunStatusSensorWarningDialog}
+          title="Run status sensor"
+          canOutsideClickClose
+          canEscapeKeyClose
+          onClose={() => {
+            setShowRunStatusWarningDialog(false);
+          }}
+          style={{width: '700px'}}
+        >
+          <DialogBody>
+            <Box margin={{bottom: 16}}>
+              This run status sensor will resume processing runs from{' '}
+              <TimeFromNow unixTimestamp={last_processed_timestamp! / 1000} />.
+            </Box>
+            To instead resume processing runs starting from now, clear the cursor before starting
+            the sensor.
+          </DialogBody>
+          <DialogFooter topBorder>
+            <Button
+              onClick={() => {
+                setShowRunStatusWarningDialog(false);
+              }}
+            >
+              Close
+            </Button>
+            <Button
+              onClick={() => {
+                onChangeSwitch();
+                setShowRunStatusWarningDialog(false);
+              }}
+            >
+              Start sensor without clearing
+            </Button>
+            <Button
+              intent="primary"
+              onClick={() => {
+                clearCursor();
```

Comment:
- Await this?
- Show a disabled state on the button while the mutation is in flight, then close the dialog?